juriscraper/oral_args/united_states/federal_appellate/ca2.py

```diff
@@ -39,11 +57,66 @@ def _download(self, request_dict={}):
         r = self.request["session"].post(self.url, params=self.parameters)
         return html.fromstring(r.content)
 
-    def _process_html(self):
-        for row in self.html.xpath("//table[@border='1']"):
+    def _extract_cases_from_html(self, html):
+        for row in html.xpath("//table[@border='1']"):
             link = row.xpath(".//a")[0].get("href")
-            docket = row.xpath(".//a/nobr/text()")[0]
-            name, date = row.xpath(".//td/text()")
-            self.cases.append(
-                {"docket": docket, "url": link, "name": name, "date": date}
+            docket = (
+                row.xpath(".//a/nobr/text()")[0]
+                if row.xpath(".//a/nobr/text()")
+                else None
             )
+            if row.xpath(".//td/text()"):
+                name, date = row.xpath(".//td/text()")
+            else:
+                continue
+            if docket and link and name and date:
+                # empty row in date ranges between 10-12-2023 and 11-11-2023
+                self.cases.append(
+                    {"docket": docket, "url": link, "name": name, "date": date}
+                )
+
+    def extract_next_page_url(self, html):
+        """Return the href url from "Next" pagination element
+        if it exists, otherwise return False.
+        """
+        path = '//table//tr/td[2]/font/a[contains(., "Next")]/@href'
+        elements = html.xpath(path)
+        return (
+            f"https://ww3.ca2.uscourts.gov{elements[0]}" if elements else False
+        )
+
+    def _get_case_dates(self):
+        dates = []
+        for case in self.cases:
+            if case["date"] == "22-2649":
+                # Incorrect data in column range between 06-14-2023 and 06-30-2023
+                case["date"] = "6-14-23"
+                case["docket"] = "22-2649"
+            dates.append(convert_date_string(case["date"]))
+        return dates
```

Comment:
can drop all of this.

Reply:
also - this bad data you are trying to fix - actually links to the wrong file - so we just want to skip it because its not actually a fix for this row